rust-webvr/src/api/vrexternal/service.rs

```diff
@@ -0,0 +1,42 @@
+use super::display::{VRExternalDisplay, VRExternalDisplayPtr};
+use super::VRExternalShmemPtr;
+use {VRDisplayPtr, VREvent, VRGamepadPtr, VRService};
+
+pub struct VRExternalService {
+    shmem: VRExternalShmemPtr,
+    display: VRExternalDisplayPtr,
+}
+
+unsafe impl Send for VRExternalService {}
+
+impl VRService for VRExternalService {
+    fn initialize(&mut self) -> Result<(), String> {
+        Ok(())
+    }
+
+    fn fetch_displays(&mut self) -> Result<Vec<VRDisplayPtr>, String> {
+        Ok(vec![self.display.clone()])
```

Comment:
It would be safer to block until `enumerationCompleted` is true to return the display. This is what Gecko does to return the display:

```
if (mDisplay == nullptr) {
      PullState(&displayState, nullptr, nullptr, [&](){
        return mEnumerationCompleted;
      });
      mDisplay = new VRDisplayExternal(displayState);
}
```